tasks/main.yml

```diff
@@ -132,6 +132,17 @@
   when:
     - deploydrupal_drush_deploy
 
+# Config Structure Check - This is primarily to catch configuration structure
+# changes from updating modules with Dependabot or missed configuration. This
+# works because the exit code for grep is non-zero when the supplied search
+# string is not found.
+- name: config structure check
+  shell: "{{ deploydrupal_drush_path }} config:export -y 3>&1 1>&2 2>&3 | grep '{{ deploydrupal_config_check_structure_string }}'"
```

Comment:
Perhaps we should add a comment above this line explaining this `3>&1 1>&2 2>&3` piping voodoo?

Reply:
I was thinking the same thing. I'm going to give the config value checking one more shot then polish this up.

Reply:
:medal_military: for redirection above and beyond the call of duty

---

Am I right to think this is meant to combine `stdout` and `stderr`?

We may or may not want to bother, but we could also do this operation in [a way that's both more "Ansibley" and more readable](https://docs.ansible.com/ansible/latest/user_guide/playbooks_error_handling.html#controlling-what-defines-failure):

```yaml
- name: config structure check
  shell: "{{ deploydrupal_drush_path }} config:export -y }}'"
  args:
    chdir: "{{ deploydrupal_core_path }}/sites/{{ deploydrupal_site_name }}"
  register: deploydrupal_config_check_result
  when:
    - deploydrupal_config_check_structure
  failed_when: >
    deploydrupal_config_check_structure_string in deploydrupal_config_check_result.stdout
    or deploydrupal_config_check_structure_string in deploydrupal_config_check_result.stderr
```
See the [`shell` module's return values](https://docs.ansible.com/ansible/latest/modules/shell_module.html#return-values) for more info.

Reply:
Updated. Thanks for the feedback!